lib/DidJwtStrategy.js

```diff
@@ -0,0 +1,204 @@
+  try {
+    decodedToken = jwt.decode(token, {complete: true});
+    payload = decodedToken.payload;
+    header = decodedToken.header;
+  } catch(err) {
+    return self.error(err);
+  }
+
+  if(!('sub' in payload)) {
+    return self.error(new BedrockError(
+      'Token `sub` is required.', 'InvalidToken', {
+        token: decodedToken
+      }));
+  }
+  const did = payload.sub;
+  if(did.indexOf('did:') !== 0) {
+    return self.error(new BedrockError(
+      'Token `sub` must be a valid decentralized identifier.', 'InvalidToken', {
+        token: decodedToken
+      }));
+  }
+
+  let verify;
+  if(header.alg.substring(0, 2) === 'RS') {
+    verify = self.verifyRsa;
+  } else if(header.alg.substring(0, 2) === 'HS' && self.store) {
+    verify = self.store.verify.bind(self.store);
+  } else {
+    return self.error(new BedrockError(
+      'Unsupported signing algorithm.', 'InvalidToken', {
+        token: decodedToken
+      }));
+  }
+
+  verify(token, err => {
+    if(err) {
+      return self.error(err);
+    }
+    req.user = {
+      identity: {id: did}
+    };
+    self.success(req.user);
+  });
+};
+
+Strategy.prototype.setStore = function(store, callback) {
+  this.store = store;
+  async.parallel([
+    callback => {
+      // the store does not handle verification of RSA signed tokens
+      const opts = bedrock.util.clone(config['authn-did-jwt'].namespaces.rsa);
+      delete opts.verify;
+      store.provision('RSA', opts, callback);
+    },
+    callback => store.provision(
+      'HMAC', config['authn-did-jwt'].namespaces.hmac, callback)
+  ], callback);
+};
+
+/**
+ * Verify a JWT signed with a RSA algorithm.
+ *
+ * @param token the token to be verified.
+ *
+ * @param callback(err, tokenPayload) called once the operation completes.
+ */
+Strategy.prototype.verifyRsa = function(token, callback) {
+  let decodedToken;
+  let header;
+
+  try {
+    decodedToken = jwt.decode(token, {complete: true});
+    header = decodedToken.header;
+  } catch(e) {
+    return callback(e);
+  }
+
+  if(!header.kid || !(/^(did:|https:)/.test(header.kid))) {
+    return callback(new Error('Invalid key identifier in token.'));
+  }
+  return async.auto({
+    // Dereference the `kid` URL and find the public key
+    getKey: callback => {
+      if((/^https:/.test(header.kid))) {
+        return request.get(header.kid, (err, res, body) =>
+          callback(err, body));
+      }
+      // kid is a DID URL
+      didio.get(header.kid, callback);
+    },
+    // If publicKey.owner is not `sub` of the JWT, get the DDO for `sub`
+    // and make sure publicKey.owner matches DDO’s `idp` entry and that
+    // it is in the whitelist
+    getDdo: ['getKey', (callback, results) => {
+      // `sub` is guaranteed to be a DID
+      if(results.getKey.owner === decodedToken.payload.sub) {
+        // FIXME: what happens in this case?  Can the sub own the key signing
+        // the token?
+        return callback(new Error('What to do!?'));
```

Comment:
We need to get the owner document and check to make sure it has the key. But then, yes, the sub can own the key -- they've proved ownership over the DID if they have signed with a key in the DDO.